packages/jetbrains-plugin/src/main/kotlin/com/mongodb/jbplugin/inspections/impl/NamespaceCheckInspectionBridge.kt

```diff
@@ -0,0 +1,140 @@
+/**
+ * This inspection is used for type checking. It also warns if a field is referenced in a
+ * query but doesn't exist in the MongoDB schema.
+ */
+
+package com.mongodb.jbplugin.inspections.impl
+
+import com.intellij.codeInspection.ProblemHighlightType
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.database.dataSource.LocalDataSource
+import com.intellij.psi.PsiElement
+import com.mongodb.jbplugin.accessadapter.datagrip.DataGripBasedReadModelProvider
+import com.mongodb.jbplugin.accessadapter.datagrip.adapter.isConnected
+import com.mongodb.jbplugin.dialects.DialectFormatter
+import com.mongodb.jbplugin.i18n.InspectionsAndInlaysMessages
+import com.mongodb.jbplugin.inspections.AbstractMongoDbInspectionBridge
+import com.mongodb.jbplugin.inspections.MongoDbInspection
+import com.mongodb.jbplugin.inspections.quickfixes.OpenConnectionChooserQuickFix
+import com.mongodb.jbplugin.linting.NamespaceCheckWarning
+import com.mongodb.jbplugin.linting.NamespaceCheckingLinter
+import com.mongodb.jbplugin.mql.Node
+import kotlinx.coroutines.CoroutineScope
+
+/**
+ * @param coroutineScope
+ */
+@Suppress("MISSING_KDOC_TOP_LEVEL")
+class NamespaceCheckInspectionBridge(coroutineScope: CoroutineScope) :
+    AbstractMongoDbInspectionBridge(
+        coroutineScope,
+        NamespaceCheckingLinterInspection,
+    )
+
+/**
+ * This inspection object calls the linting engine and transforms the result so they can be rendered in the IntelliJ
+ * editor.
+ */
+internal object NamespaceCheckingLinterInspection : MongoDbInspection {
```

Comment:
What is the convention for declaring objects in Kotlin? Are they camel cased or pascal cased?

Reply:
Objects are like static classes, so usually they are pascal cased 👀 .